book/src/guide/specs.md

```diff
@@ -73,47 +73,59 @@ An example of refining an extern function can be found
 To define an extern spec on a function, you need to do three things, which
 happen to correspond to each of the below lines.
 
-```
-#[extern_spec(std::mem)]
-#[flux_rs::sig(fn(&mut i32[@a], &mut i32{v : a < v }) -> ())]
-fn swap(a: &mut i32, b: &mut i32);
+```rust
+{{#include ../../../tests/tests/neg/surface/extern_spec_swap.rs:5:7}}
 ```
 
 1. Add the `#[extern_spec]` attribute. This attribute optionally takes a path;
    in the above example, this is `std::mem`. You can use this path to qualify
    the function. So in the above example, the function we are targeting has the
    full path of `std::mem::swap`.
-2. Add a `#[flux_rs::sig(...)]` attribute. This is required for any extern spec on
-   a function. This signature behaves as if the `#[flux_rs::trusted]` attribute was
-   added, because we can't actually check the implementation. We just verify
-   some simple things, like that the function arguments have compatible types.
-3. Write a function stub that matches the external function.
+2. Add a `#[spec(...)]` (or equivalently `#[flux_rs::sig(...)]`) attribute, which
+   is required for any extern spec on a function. This signature behaves as if
+   the `#[flux_rs::trusted]` attribute was added, because we cannot _actually_
+   check the implementation. Instead, flux just verifies some simple things,
+   like that the function arguments have compatible types.
+3. Write a function stub whose rust signature matches the external function.
 
 If you do the above, you can use `std::mem::swap` as if it were refined by the
 above type.
 
-You shouldn't need to know the details, but here's how the macro works. It
-parses the `std::mem` into a module path and then transforms the function into
 
+```rust
+{{#include ../../../tests/tests/neg/surface/extern_spec_swap.rs:9:16}}
 ```
-#[flux_rs::extern_spec]
-#[flux_rs::sig(fn(&mut i32[@a], &mut i32{v : a < v }) -> ())]
+
+You shouldn't need to know the details, but if you're curious,
+here's how the macro works. In the code above, flux parses the
+`std::mem` into a module path and then transforms the function into
+
+```rust
+#[extern_spec]
+#[spec(fn(x: &mut T[@vx], y: &mut T[@vy]) ensures x: T[vy], y: T[vx])]
```

Comment:
This is not exactly true; the macro expands to something different now. We can probably remove this part.